src/server.ts

```diff
@@ -0,0 +1,105 @@
+import express, { json, Request, Router } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import * as Joi from 'joi';
+import { ContainerTypes, ValidatedRequest, ValidatedRequestSchema, createValidator } from 'express-joi-validation';
+
+interface IUser {
+    id: string;
+    login: string;
+    password: string;
+    age: number;
+    isDeleted: boolean;
+}
+
+interface QueryParams {
+    loginSubstring: string;
+    limit: string;
+}
+
+interface IGetAutoSuggestUsers {
+    loginSubstring?: string;
+    limit?: number;
+}
+
+const schema = Joi.object({
+    login: Joi.string().required(),
+    password: Joi.string().alphanum().required(),
+    age: Joi.number().min(4).max(130).required()
+});
+
+interface RequestSchema extends ValidatedRequestSchema {
+    [ContainerTypes.Body]: {
+        login: string;
+        password: string;
+        age: number;
+    };
+}
+
+const validator = createValidator();
+
+const router = Router();
+
+const app = express();
+
+const getAutoSuggestUsers = ({ loginSubstring = '', limit = 100 }: IGetAutoSuggestUsers): IUser[] => {
+    return users
+        .filter((u) => u.login.startsWith(loginSubstring))
+        .sort((a, b) => a.login.localeCompare(b.login))
+        .slice(0, limit);
+};
+
+const users: IUser[] = [];
+
+router
+    .route('/users')
+    .post(validator.body(schema), (req: ValidatedRequest<RequestSchema>, res) => {
```

Comment:
it is a good idea to separate router and logic within controllers,
you can move controllers in the separate file, so your code on router level will be as following:"
`router

> .post(validator, postController)

> .get(getController)

`